parser/src/main/java/parser/publio/NextPageUrlConsumer.java

```diff
@@ -0,0 +1,63 @@
+package parser.publio;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jsoup.nodes.Document;
+import parser.PageLoader;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * This thread is needed for consuming url to next pages to produce urls to books details
+ */
+class NextPageUrlConsumer implements Runnable {
+
+    private final Logger logger = LogManager.getRootLogger();
+
+    BlockingQueue<String> urlToNextPageQueue;
+    PublioUrlParser publioUrlParser;
+    PageLoader pageLoader;
+    List<String> bookDetailsUrls;
+
+    public NextPageUrlConsumer(BlockingQueue urlToNextPageQueue,PageLoader pageLoader) {
+        this.urlToNextPageQueue = urlToNextPageQueue;
+        this.pageLoader = pageLoader;
+        this.publioUrlParser = new PublioUrlParser();
+        bookDetailsUrls = new ArrayList<>();
+    }
+
+    List<String> getListToBookDetails(String page) {
+        List<String> pages = new ArrayList<>();
+        Optional<Document> document = pageLoader.getPage(page);
+        pages.addAll(publioUrlParser.getLinksToBooksDetails(document));
+        return pages;
+    }
+
+    @Override
+    public void run() {
+        logger.debug("start thread");
```

Comment:
You should add some thread name to this log and also to whole class. I know that there is someone who will punish you with chair if you do not do this